### DISCUSSION TITLE
Make text BOLD and ITALIC or BOTH when creating PDF

### DISCUSSION THREAD

**Discussion by rafayaar** [pymupdf/PyMuPDF#3956] [Looking for help] [answered]
@JorjMcKie 

**My finds on it:**
1- Font file is used to make text bold, italic or both. 
2- We can add 2 or 3 layers of same text with small offset to the Point location and it will look like a bold
3- To user insert_html

For me I have fixed repository of font files. I can keep multiple version of a Base Font. Like one for Calibri regular, then italic, bold, semi bold and so on
I also tried getting bold using offset but that looks odd.

I cant get to find anything that I can use with insert_text.

I am still searching more on insert_html, but looking for something straight forward

**Comment by JorjMcKie** [comment, marked as answer]
To start with `insert_htmlbox`:
* There is no parameter **_in the method_** directly to use a certain font of desired characteristics. You must use HTML / CSS syntax to select a font of your liking. As for the standard variations, you can simply specify the HTML construct `<b>bold text</b>`. The MuPDF Story behind the method will locate a suitable / available font. Same is true for semi-bold etc.: use CSS syntax to define a `@font-face`.
* As is true for **_all_** text insertions, outputting **_exactly_** the same text and a rectangle with exactly the same width and height, plus a small (!) shift will simulate bold or even character shading.

In addition to the previous, `Page.insert_text()`/`Page.insert_textbox()` also support the `border_width` and the `render_mode` parameters which generates the PDF command `Tr` (text rendering). The default value `0 Tr` writes the character interiors (only, i.e. "fill"), `1 Tr` writes character borders (i.e. "stroke"), and `2 Tr` writes both.
Combining both parameters, achieves a number of effects like documented [here](https://pymupdf.readthedocs.io/en/latest/shape.html#commonparms).

Finally, you can use the `morph=(point, matrix)` parameter to simulate more effects like italics and shrinking / stretching by using `point` to be the insertion point in `insert_text(point, ...)` and a suitable matrix.

Here is an example script for some of these things.
[font-effects.zip](https://github.com/user-attachments/files/17410604/font-effects.zip)

**Reply by rafayaar** [reply]
Thanks. It helped
I talked to few people regarding font families. I was unawared that for Bold italic and other styles, different font files are used of same family. I mean using Calibri-Regular, we cant get Bold or Italic text. Then I wanted to explore if there is any way that a single file can contain mapping against all styles.
Couldn't find anything

I have concluded that, font styles and weights are properties of Font and not Text.

**Reply by JorjMcKie** [reply]
Correct findings! The usual variants (regular, bold, italic, bold-italic) require 4 different font files. Inevitably.
With an appropriate CSS definition of the font family, the selection of the right variant is automatic whenever you use the respective HTML constructs `<b>`, `<i>` etc.
